File: packages/sdk/src/invites.ts
import {QelosSDKOptions} from './types';
import BaseSDK from './base-sdk';

export interface IInvite {
  workspace: {
    name: string;
    logo?: string;
    _id: string;
  }
}

export enum InviteKind {
  DECLINE = 'decline',
  ACCEPT = 'accept',
}

export default class QlInvites extends BaseSDK {
  private relativePath = '/api/invites';

  constructor(options: QelosSDKOptions) {
    super(options)
  }

  getList() {
    return this.callJsonApi<IInvite[]>(this.relativePath);
  }

  acceptWorkspace(workspaceId: string): Promise<unknown> {
    return this.callApi(
      `${this.relativePath}/${workspaceId}`,
      {
        method: 'put',
        headers: {'content-type': 'application/json'},
        body: JSON.stringify({
          workspace: workspaceId,
          kind: InviteKind.ACCEPT
        })
      }
    )
  }

  declineWorkspace(workspaceId: string): Promise<unknown> {
    return this.callApi(
      `${this.relativePath}/${workspaceId}`,
      {
        method: 'put',
        headers: {'content-type': 'application/json'},
        body: JSON.stringify({
          workspace: workspaceId,
          kind: InviteKind.DECLINE
        })
      }
    )
  }
}

File: packages/python-sdk/qelos_sdk/invites.py
from __future__ import annotations

import json
from typing import Any, Dict, List

from .base_sdk import BaseSDK
from .types import QelosSDKOptions


class InviteKind:
    DECLINE = "decline"
    ACCEPT = "accept"


class QlInvites(BaseSDK):
    """Workspace invite management: list, accept, and decline invites."""

    _relative_path = "/api/invites"

    def __init__(self, options: QelosSDKOptions) -> None:
        super().__init__(options)

    async def get_list(self) -> List[Dict[str, Any]]:
        """List pending invites for the current user."""
        return await self.call_json_api(self._relative_path)

    async def accept_workspace(self, workspace_id: str) -> Any:
        """Accept a workspace invite."""
        return await self.call_api(
            self._relative_path,
            method="POST",
            headers={"content-type": "application/json"},
            body=json.dumps({"workspace": workspace_id, "kind": InviteKind.ACCEPT}),
        )

    async def decline_workspace(self, workspace_id: str) -> Any:
        """Decline a workspace invite."""
        return await self.call_api(
            self._relative_path,
            method="POST",
            headers={"content-type": "application/json"},
            body=json.dumps({"workspace": workspace_id, "kind": InviteKind.DECLINE}),
        )
